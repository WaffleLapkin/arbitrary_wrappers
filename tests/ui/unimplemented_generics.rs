#![feature(arbitrary_self_types)]
use arbitrary_wrappers::use_ast;

// TODO: make the error cleaner

fn main() {
    assert_eq!(Wrapper(Type, 0).method(), 0);
}

struct Wrapper<T, U>(T, U);

struct Type;

#[use_ast(Type)]
impl<U> Wrapper<Type, U> {
    fn method(self) -> U {
        self.1
    }
}

impl<T, U> core::ops::Deref for Wrapper<T, U> {
    type Target = T;

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}
