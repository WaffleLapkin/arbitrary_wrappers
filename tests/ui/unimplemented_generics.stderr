error: custom attribute panicked
  --> $DIR/unimplemented_generics.rs:14:1
   |
14 | #[use_ast]
   | ^^^^^^^^^^
   |
   = help: message: not implemented: Generics and lifetimes are currently not supported (yet?). Please fill an issue at `https://github.com/wafflelapkin/arbitrary_wrappers` describing your use case.

error[E0599]: no method named `method` found for struct `Wrapper<Type, {integer}>` in the current scope
  --> $DIR/unimplemented_generics.rs:7:33
   |
7  |     assert_eq!(Wrapper(Type, 0).method(), 0);
   |                                 ^^^^^^ method not found in `Wrapper<Type, {integer}>`
...
10 | struct Wrapper<T, U>(T, U);
   | --------------------------- method `method` not found for this
